### PR TITLE
simplify tx confirmations display

### PR DESCRIPTION
**Pull request by derhuerst**
This PR makes `TxHash` look a lot less verbose, aligning the UI more with targeted user group. Advanced users still get the block number when hovering.

I added a new ~~`ui/Hash`~~ `ui/ShortenedHash` component, which is supposed to be put in every place where a tx hash is shown. Will also expand this to address hashes.

Except the obvious changes, there is a small visual regression in this PR: The tx hash in the Signer is now so short that the left-alignment of the text is visible.

signer-regression

### CODE REVIEW

**Comment by derhuerst** [comment]
Also thought about using a non-linear scale to visualise the change of a tx getting rejected by the network properly.

File: js/src/ui/ShortenedHash/shortenedHash.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export default class ShortenedHash extends Component {
+  static propTypes = {
+    data: PropTypes.string.isRequired
+  }
+
+  render () {
+    const { data } = this.props;
+
+    let shortened = data.toLowerCase();
+    if (shortened[0] === '0' && shortened[1] === 'x') {
```
Comment (jacogr) [inline js/src/ui/ShortenedHash/shortenedHash.js]: any reason for not just checking `shortened.substr(0, 2) === '0x'` (as done everywhere else - consistency so we don't need to think about how xyz does it)

**Comment by jacogr** [comment]
Good, right step. Like you suggested, would probably just fix up the Signer display a little bit so it looks presentable. (Alternatively just log an issue for it, I have all the Signer items in my queue, so busy reworking the full display & adding the required additional functionality)

File: js/src/ui/TxHash/txHash.js [hunk 2]
```diff
@@ -87,16 +90,28 @@ class TxHash extends Component {
...
-      txBlock = `#${transaction.blockNumber.toFormat(0)}`;
-      confirmations = num.toFormat(0);
-      value = num.gt(maxConfirmations) ? maxConfirmations : num.toNumber();
+    const confirmations = blockNumber.minus(transaction.blockNumber).plus(1);
+    const value = Math.min(confirmations.toNumber(), maxConfirmations);
+    let count;
+    if (confirmations.gt(maxConfirmations)) {
+      count = confirmations.toFormat(0);
+    } else {
+      count = confirmations.toFormat(0) + '/' + maxConfirmations;
```
Comment (derhuerst) [inline js/src/ui/TxHash/txHash.js]: You will always be able to argue with the point of consistency and I agree that it is important. Readable code is more important though imo.

I find

```js
foo + '/' + bar
```

a lot less noisy than

```js
`${a}/${b}`
```

Will adapt to whatever you decide, but keep this in mind.

**Comment by derhuerst** [comment]
> Good, right step. Like you suggested, would probably just fix up the Signer display a little bit so it looks presentable. (Alternatively just log an issue for it, I have all the Signer items in my queue, so busy reworking the full display & adding the required additional functionality)

I'd propose to have a different way of showing that a TX is confirmed/rejected/pending. Just putting it in a column the importance of the information. I suggest we decide on a new look for the Signer as a group.

Until that happened, I'll center the "Rejected"/"12 Confirmations" messages.

File: js/src/ui/TxHash/txHash.js (hunk 2 above)
Comment (ngotchac) [inline js/src/ui/TxHash/txHash.js]: I would be balances between the two of you. IMO:

```
count = `${confirmations}/${maxConfirmations}`
```
and
```
count = confirmations.toFormat(0) + `/${maxConfirmations}`
```

Ie. if just variables, then string template, if returned value of method, `+` concat.

**Comment by jacogr** [comment]
Something looks weird here - it contains files from other PRs, not sure what was done here and what not.

**Comment by derhuerst** [comment]
> Something looks weird here - it contains files from other PRs, not sure what was done here and what not.

The commits that are actually part of this PR (and not merged in) are the following:

1. 97db6e65296ab20b22bc6790d73767a6bef1bf3a
2. 0d832cbd62e52c571b4080c20d07e5fe27fbe82d
3. 35c265f668ffdd7a69a3392e5fd29d32a909a46f
4. afc068770dba32033a11838e0452cbffcfb2efe7
5. 5bc6ed556ebfcce6964453d28866353de3f0a640
6. 85227f9abbaabb37f5f6fd3bd7f313f6ada27e11
7. 14984c3f64f71525347d7647cabd1393ed23c8bf

**Pull request merged**